src/pages/GuestInfo.tsx

```diff
@@ -0,0 +1,159 @@
+import { Box, FormField, TextInput, DateInput, Form, Button } from 'grommet';
+import { useNavigate } from 'react-router-dom';
+import { PageWrapper } from './PageWrapper';
+import Logger from '../utils/logger';
+import { useCallback, useState } from 'react';
+import axios from 'axios';
+import { MessageBox } from 'src/components/MessageBox';
+import { DateTime } from 'luxon';
+import { useAppDispatch, useAppState } from '../store';
+import { PersonalInfo } from 'src/store/types';
+import { backend } from '../config';
+
+// const offerId = 1;
+const logger = Logger('GuestInfo');
+const defaultValue = {
+  firstname: '',
+  lastname: '',
+  birthday: new Date(),
+  email: '',
+  phone: ''
+};
+
+export const GuestInfo = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { checkout } = useAppState();
+
+  const [value, setValue] = useState<PersonalInfo>(defaultValue);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<undefined | string>();
+
+  const handleSubmit = useCallback(async () => {
+    try {
+      setError(undefined);
+      setLoading(true);
+      if (
+        value.firstname === '' ||
+        value.lastname === '' ||
+        value.email === '' ||
+        value.phone === ''
+      ) {
+        throw Error('Empty fields');
```

Comment:
will the user also see feedback if this error is being thrown?

Reply:
@bobetbat Please use MessageBox for general errors, but for errors in the context of fields, you have to use built-in Grommet mechanism (validation => error).

Reply:
✔️